### ISSUE TITLE
Replace reccuring terms in dydx to speed up compilation

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/AMICI#119]
Some optimization could be done in places like:
```
  dydx[1+115*50] = 1.442695040888963/(x_tmp[115]+x_tmp[133]+x_tmp[141]+x_tmp[151]+x_tmp[161]*2.0);
  dydx[1+133*50] = 1.442695040888963/(x_tmp[115]+x_tmp[133]+x_tmp[141]+x_tmp[151]+x_tmp[161]*2.0);
  dydx[1+141*50] = 1.442695040888963/(x_tmp[115]+x_tmp[133]+x_tmp[141]+x_tmp[151]+x_tmp[161]*2.0);
  dydx[1+151*50] = 1.442695040888963/(x_tmp[115]+x_tmp[133]+x_tmp[141]+x_tmp[151]+x_tmp[161]*2.0);
  dydx[1+161*50] = 2.885390081777927/(x_tmp[115]+x_tmp[133]+x_tmp[141]+x_tmp[151]+x_tmp[161]*2.0);
```
by evaluating the reccuring terms only once.

This is already done for `_xdot.cpp` in `_w.cpp`, see `getSyms.m`.

**Comment by dweindl** [comment]
This will be covered by #213.
